_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-external-media.php

```diff
@@ -0,0 +1,421 @@
+				),
+			),
+		),
+	);
+
+	/**
+	 * Service regex.
+	 *
+	 * @var string
+	 */
+	private static $services_regex = '(?P<service>google_photos|pexels)';
+
+	/**
+	 * Temporary filename.
+	 *
+	 * Needed to cope with Google's very long file names.
+	 *
+	 * @var string
+	 */
+	private $tmp_name;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'external-media';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Registers the routes for external media.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/list/' . self::$services_regex,
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_external_media' ),
+				'permission_callback' => array( $this, 'permission_callback' ),
+				'args'                => array(
+					'search'      => array(
+						'description' => __( 'Media collection search term.', 'jetpack' ),
+						'type'        => 'string',
+					),
+					'number'      => array(
+						'description' => __( 'Number of media items in the request', 'jetpack' ),
+						'type'        => 'number',
+						'default'     => 20,
+					),
+					'path'        => array(
+						'type' => 'string',
+					),
+					'page_handle' => array(
+						'type' => 'string',
+					),
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/copy/' . self::$services_regex,
+			array(
+				'methods'             => \WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'copy_external_media' ),
+				'permission_callback' => array( $this, 'permission_callback' ),
+				'args'                => array(
+					'media' => array(
+						'description'       => __( 'Media data to copy.', 'jetpack' ),
+						'items'             => array_values( $this->media_schema ),
+						'required'          => true,
+						'type'              => 'array',
+						'sanitize_callback' => array( $this, 'sanitize_media' ),
+						'validate_callback' => array( $this, 'validate_media' ),
+					),
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/connection/(?P<service>google_photos)',
+			array(
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_connection_details' ),
+				'permission_callback' => array( $this, 'permission_callback' ),
+			)
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to external media libraries.
+	 */
+	public function permission_callback() {
+		return current_user_can( 'edit_posts' );
+	}
+
+	/**
+	 * Sanitization callback for media parameter.
+	 *
+	 * @param array $param Media parameter.
+	 * @return true|\WP_Error
+	 */
+	public function sanitize_media( $param ) {
+		$param = $this->prepare_media_param( $param );
+
+		return rest_sanitize_value_from_schema( $param, $this->media_schema );
```

Comment:
Noting that this is currently triggering PHP notices:

```
PHP Notice:  Undefined index: type in /var/www/html/wp-includes/rest-api.php on line 1482
```

Reply:
@jeherve just to be sure, are you fine merging now and addressing the notice separately or should we do it now?

Reply:
I think we can merge now and address this later 👍